src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -520,6 +529,45 @@ public function iShouldNotSeeShippingTotal(): void
         Assert::same($this->responseChecker->getValue($this->client->getResponse(), 'shippingTotal'), 0);
     }
 
+    /**
+     * @Then /^I should(?:| also) be notified that the "([^"]+)" and the "([^"]+)" in (shipping|billing) details are required$/
+     */
+    public function iShouldBeNotifiedThatTheAndTheInShippingDetailsAreRequired($firstElement, $secondElement, $detailType)
+    {
+
+        /** @var Response|null $response */
+        $response = $this->client->getResponse();
+
+        Assert::true($response->getStatusCode() === 400);
+
+        /** @var array|null $violations */
+        $violations = $this->responseChecker->getResponseContent($response)['violations'];
+
+        $detailType .= 'Address';
+
+        $hasFirstElement = false;
+        $hasSecondElement = false;
+
+        $firstElement = $detailType . '.' . u($firstElement)->camel();
+        $secondElement = $detailType . '.' . u($secondElement)->camel();
+
+        foreach($violations as $violation) {
+
+            if($hasFirstElement === false && str_contains($firstElement, $violation['propertyPath'])) {
+                $hasFirstElement = true;
+            }
+            if($hasSecondElement === false && str_contains($secondElement, $violation['propertyPath'])) {
+                $hasSecondElement = true;
+            }
+            if ($hasFirstElement === true && $hasSecondElement === true) {
+                break;
+            }
+        }
```

Comment:
```suggestion
        foreach($violations as $violation) {
            if ($hasFirstElement === false && str_contains($firstElement, $violation['propertyPath'])) {
                $hasFirstElement = true;
            }
            if ($hasSecondElement === false && str_contains($secondElement, $violation['propertyPath'])) {
                $hasSecondElement = true;
            }
            if ($hasFirstElement === true && $hasSecondElement === true) {
                break;
            }
        }
```
But it could be extracted to a separate private method to check if there is a violation (and its message) for a given element